localstack/services/cloudformation/deployment_utils.py

```diff
@@ -135,3 +135,20 @@ def select_parameters(*param_names):
 
 def is_none_or_empty_value(value):
     return not value or value == PLACEHOLDER_AWS_NO_VALUE
+
+
+def generate_default_name(stack_name: str, logical_resource_id: str):
+    random_id_part = short_uid()
+    resource_id_part = logical_resource_id[:24]
+    stack_name_part = stack_name[: 63 - 2 - (len(random_id_part) + len(resource_id_part))]
+    return f"{stack_name_part}-{resource_id_part}-{random_id_part}"
+
+
+def pre_create_default_name(key: str):
```

Comment:
nit: could add a type hint to indicate that this returns a `Callable[...]` (although the parameters of the callable will likely soon be simplified/unified).